File: solutions/3159. Find Occurrences of an Element in an Array/3159.py
# Time complexity: O(n+q)
# Space complexity: O(n+q)
class Solution:
  def occurrencesOfElement(self, nums: List[int], queries: List[int], x: int) -> List[int]:
    indices = [i for i, num in enumerate(nums) if num == x]
    return [indices[query - 1] if query <= len(indices) else -1
            for query in queries]

File: solutions/3159. Find Occurrences of an Element in an Array/3159.cpp
// Time complexity: O(n+q)
// Space complexity: O(n+q)
class Solution {
 public:
  vector<int> occurrencesOfElement(const vector<int>& nums, 
                                   const vector<int>& queries, int x) {
    const vector<int> indices = getIndices(nums, x);
    vector<int> ans;

    for (const int query : queries)
      ans.push_back(query <= indices.size() ? indices[query - 1] : -1);

    return ans;
  }

 private:
  vector<int> getIndices(const vector<int>& nums, int x) {
    vector<int> indices;
    for (int i = 0; i < nums.size(); ++i)
      if (nums[i] == x)
        indices.push_back(i);

    return indices;
  }
};